tests/subscription/test_rhsm.py

```diff
@@ -1,6 +1,361 @@
+
+        # check repo status in guest
+        _, output = ssh_guest.runcmd('subscription-manager repos --list')
+        assert msg_search(output, 'Available Repositories')
+
+        # check subscription status in guest
+        _, output = ssh_guest.runcmd('subscription-manager status')
+        assert (msg_search(output, 'Overall Status: Current')
+                and not msg_search(output, 'Invalid'))
+
+    @pytest.mark.tier1
+    @pytest.mark.notRemoteMode
+    def test_vdc_physcial_pool_consumed_status_in_physical_host(
+            self, sm_host, vdc_pool_physical
+    ):
+        """Test vdc physcial pool consumed status in physical host when
+            set cpu_socket(s).
+
+        :title: virt-who: rhsm: test physcial vdc pool consumed status in
+            physical host
+        :id: cbf0e07b-c51c-4e6d-9c80-07c2c8dd7692
+        :caseimportance: High
+        :tags: tier1
+        :customerscenario: false
+        :upstream: no
+        :steps:
+
+            1. Create facts with 'cpu.cpu_socket(s): 4' for physical host
+            2. Attach physcial vdc pool for local host with quantity=1
+            3. Check the consumed status for host
+            4. Remove the consumed pool from host
+            5. Attach physcial vdc pool for local host without quantity setting
+            6. Check the consumed status for host
+            7. Remove the custom facts setting
+
+        :expectedresults:
+
+            1. With cpu_socket(s)=4, the physical host can subscribe only one
+                physical vdc pool with status 'Only supports 2 of 4 sockets.'
+                because 1 pool can cover 2 sockets
+            2. Without quantity setting, the physcial host default to subscribe
+                2 vdc pool with status 'Subscription is current'
+        """
+        try:
+            sm_host.facts_create('cpu.cpu_socket(s)', '4')
+            sm_host.unattach()
+            # Attach physcial vdc pool with quantity=1
+            sm_host.attach(pool=vdc_pool_physical, quantity=1)
+            consumed_data = sm_host.consumed(sku_id=vdc_physical_sku,
+                                             sku_type='Physical')
+            assert (consumed_data['quantity_used'] == '1'
+                    and consumed_data[
+                        'status_details'] == 'Only supports 2 of 4 sockets.')
+
+            # Attach physcial vdc pool without quantity setting
+            sm_host.unattach()
+            sm_host.attach(pool=vdc_pool_physical)
+            consumed_data = sm_host.consumed(sku_id=vdc_physical_sku,
+                                             sku_type='Physical')
+            assert (consumed_data['quantity_used'] == '2'
+                    and consumed_data[
+                        'status_details'] == 'Subscription is current')
+
+        finally:
+            sm_host.facts_remove()
+
+    @pytest.mark.tier1
+    def test_vdc_virtual_pool_consumed_status_in_guest(
```

Comment:
will only test the vdc consumed status for rhsm, not fro satellite, because it's not the virt-who function, only to test the vdc sku properties.